zipline/src/jniTest/kotlin/app/cash/zipline/EndpointTest.kt

```diff
@@ -163,4 +164,65 @@ internal class EndpointTest {
     assertThat(thrownException).hasMessageThat()
       .isEqualTo("java.lang.IllegalStateException: boom!")
   }
+
+  @Test
+  fun handleBridgesServicesBeforeEndpointsAreConnected() {
+    val requests = LinkedBlockingDeque<String>()
+    val responses = LinkedBlockingDeque<String>()
+    val service = object : EchoService {
+      override fun echo(request: EchoRequest): EchoResponse {
+        requests += request.message
+        return EchoResponse(responses.take())
+      }
+    }
+
+    val handleA = Handle<EchoService>(EchoSerializersModule, service)
+    (handleA as InboundHandle<*>).connect(endpointA, "helloService")
+
+    // Note that we cast OutboundHandle<EchoService> down to Handle<EchoService> before calling
+    // get(). This is necessary because we don't bother to rewrite calls to overrides.
+    val handleB: Handle<EchoService> = OutboundHandle()
+    (handleB as OutboundHandle<EchoService>).connect(endpointB, "helloService")
+
+    val client = handleB.get(EchoSerializersModule)
+
+    responses += "this is a curt response"
+    val response = client.echo(EchoRequest("this is a happy request"))
+    assertEquals("this is a curt response", response.message)
+    assertEquals("this is a happy request", requests.poll())
+    assertNull(responses.poll())
+    assertNull(requests.poll())
+  }
+
+  interface EchoServiceFactory {
+    fun create(greeting: String): Handle<EchoService>
+  }
+
+  @Test
+  fun transmitHandles() {
```

Comment:
This is the big payoff. We can instantiate an interface and send handles to it without names, and without directly referencing the Zipline that they connect through.